flux/control/controller.go

```diff
@@ -40,7 +41,7 @@ func (c *Controller) Query(ctx context.Context, compiler flux.Compiler) (flux.Qu
 		ctx = dep.Inject(ctx)
 	}
 
-	p, err := compiler.Compile(ctx)
+	p, err := compiler.Compile(ctx, runtime.Default)
```

Comment:
this doesn't seem right to me, I believe we have to inject some packages here, but I am not certain.

Reply:
I think that since we are using the global injection method we should be using the global runtime.